include/cuco/detail/static_map.inl

```diff
@@ -163,7 +163,7 @@ __device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::i
   auto current_slot = initial_slot(g, insert_pair.first, hash);
 
   while (true) {
-    key_type const existing_key = current_slot->first;
+    key_type const existing_key = current_slot->first.load(cuda::memory_order_relaxed);
```

Comment:
there is one other place in this file where we use SC load on the key:
https://github.com/NVIDIA/cuCollections/blob/99d3b85d5ca7a22cb9df24b1b0f14e4b2ea4c69a/include/cuco/detail/static_map.inl#L128
should we update that as well?

Reply:
That one is actually just getting a _reference_ to the atomic object, not loading the stored value. So there's no need for a memory order there.

Reply:
got it, thanks!
Another question - do we want to make the relaxed order default somehow, since we never use any other type of ordering? I can see some arguments against it that we want to be explicit, but just feels like it's also easy to forget to set the ordering to relaxed when writing new code and possibly hit a perf issue..

Reply:
Unfortunately there isn't an easy way to do that. We're just using the `cuda::atomic` type directly from libcu++ and it's loads default to `seq_cst` same as `std::atomic`. The design philosophy was to guarantee correctness by default and make improved performance opt-in at the cost of complexity.